### PR TITLE
[development] added required runtime dependency Gem::PhysX.Editor to WhiteBox.Editor.Physics.Tests 

### PR DESCRIPTION
**Pull request by AMZN-ScottR**
Signed-off-by: AMZN-ScottR <24445312+AMZN-ScottR@users.noreply.github.com>

### CODE REVIEW

**Review by moraaar — APPROVED** [review]
LGTM

**Review by gadams3 — APPROVED** [review]

File: Gems/WhiteBox/Code/CMakeLists.txt [hunk 1]
```diff
@@ -198,6 +198,8 @@ if(PAL_TRAIT_BUILD_TESTS_SUPPORTED)
                     AZ::AzTestShared
                     AZ::AzManipulatorTestFramework.Static
                     Gem::WhiteBox.Editor.Static
+            RUNTIME_DEPENDENCIES
+                Gem::PhysX.Editor
```
Comment (lumberyard-employee-dm) [inline Gems/WhiteBox/Code/CMakeLists.txt:202]: The UnitTest that runs in the AzTestRunner only loads the dll of the test that is running.

Adding a dependency to `Gem::PhysX.Editor` will not have the `WhiteBox.Editor.Physics.Tests` loading automatically in the AzTestRunner.

The UnitTest code itself would have to load any dependent .dll/.so.

**Review by lumberyard-employee-dm — CHANGES_REQUESTED** [review]

File: Gems/WhiteBox/Code/CMakeLists.txt (hunk 1 above)
Comment (AMZN-ScottR) [reply Gems/WhiteBox/Code/CMakeLists.txt:202]: Right, the tests here already do that:
https://github.com/o3de/o3de/blob/fed1278fe64e8a435b1c19ba8092b18a98fa9eb1/Gems/WhiteBox/Code/Tests/WhiteBoxPhysicsTest.cpp#L50

**Review by hultonha — APPROVED** [review]
Thanks for fixing this! 👍

File: Gems/WhiteBox/Code/CMakeLists.txt (hunk 1 above)
Comment (lumberyard-employee-dm) [reply Gems/WhiteBox/Code/CMakeLists.txt:202]: Alright this is good then.

**Review by lumberyard-employee-dm — APPROVED** [review]

**Pull request merged**